x-pack/plugins/security_solution/cypress/e2e/detection_rules/all_rules_read_only.cy.ts

```diff
@@ -27,6 +27,9 @@ describe('All rules - read only', () => {
     cleanKibana();
     createCustomRule(getNewRule(), '1');
     login(ROLES.reader);
+  });
+
+  beforeEach(() => {
```

Comment:
Awesome! We've also noticed that there are a lot of e2e tests where the state is not properly reset between the tests. Thank you for fixing it and enforcing good practices! 👍 

As a side note (mainly for the rules area team), I'd probably merge all `it`s in this file into a single test. Not a suggestion for this PR.